packages/@rnw-scripts/doxysaurus/src/doc-model.ts

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ *
+ * @format
+ **/
+
+//
+// Documentation model.
+//
+
+// Documentation model for a code project.
+export class DocModel {
+  // The code project for the model.
+  codeProject = '';
+
+  // List of all documented compounds indexed by their name.
+  compounds: {[index: string]: DocCompound} = {};
```

Comment:
Consider `Record<string, DocCompound>`

instead of `{[index: string]: DocCompound}`

Difference is mostly stylistic. IIRC one came after the other. I think I see Record more often though.

Reply:
I tried to find more information about Record vs Indexer. So far I understand that for mapping any string to another type the indexer is still considered to be the canonical syntax. The Record is recommended for cases when we need to restrict the indexer type. I have changed this code to use array in commit https://github.com/microsoft/react-native-windows/pull/7044/commits/464f21524bddd39a72e4037dd6a20de9dec2de62